provision/kubernetes/cluster.go

```diff
@@ -660,7 +609,14 @@ func (c *ClusterClient) configForContext(context, key string) string {
 	if v, ok := c.CustomData[context+":"+key]; ok {
 		return v
 	}
-	return c.CustomData[key]
+	if v, ok := c.CustomData[key]; ok {
+		return v
+	}
+	data, err := config.GetString("clusters:defaults:" + key)
```

Comment:
@morpheu have risky to raise an exception of key not found here?

Reply:
In such case always returns an empty string. Most of methods were check also for empty CustomData, so ignoring all config errors and treat them as empty string seems ok to me